packages/data-broker/src/DataBroker.js

```diff
@@ -123,6 +177,11 @@ export class DataBroker {
 
   pullForServiceAndType = async (dataSources, options, type) => {
     const { service_type: serviceType } = dataSources[0];
+    const permissionChecker = this.permissionCheckers[type];
+    const userHasPermission = await permissionChecker(dataSources);
+    if (!userHasPermission) {
+      throw new Error(`Missing permissions for some or all ${type}s for ${serviceType}`);
```

Comment:
I suspect this error message will puzzle a lot of devs/users 😆 
If possible lets list the data elements / data groups the user doesn't have permission for.

Reply:
Added some better error messaging

Reply:
Beautiful, thank you!